components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -0,0 +1,628 @@
+    }
+
+    private Map<String, Object> getClaimMapForUser(OAuthTokenReqMessageContext requestMsgCtx,
+                                                   Map<ClaimMapping, String> userAttributes) {
+        Map<String, Object> claimMap = Collections.emptyMap();
+        if (isEmpty(userAttributes)) {
+            if (!isFederatedUser(requestMsgCtx)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No user attributes found in cache for user: " + requestMsgCtx.getAuthorizedUser() + "." +
+                            " Retrieving claims from userstore.");
+                }
+                return retrieveClaimsForLocalUser(requestMsgCtx, claimMap);
+            }
+        }
+        return getClaimsMap(userAttributes);
+    }
+
+    private Map<String, Object> retrieveClaimsForLocalUser(OAuthTokenReqMessageContext requestMsgCtx,
+                                                           Map<String, Object> claimMap) {
+        try {
+            claimMap = getClaimsFromUserStore(requestMsgCtx);
+            if (log.isDebugEnabled()) {
+                log.debug("User attributes not found in cache. Retrieved attributes for local user: " +
+                        requestMsgCtx.getAuthorizedUser() + " from userstore.");
+            }
+        } catch (UserStoreException | IdentityApplicationManagementException | IdentityException e) {
+            log.error("Error occurred while getting claims for user: " + requestMsgCtx.getAuthorizedUser() +
+                    " from userstore.", e);
+        }
+        return claimMap;
+    }
+
+    private boolean isFederatedUser(OAuthTokenReqMessageContext requestMsgCtx) {
+        return requestMsgCtx.getAuthorizedUser().isFederatedUser();
+    }
+
+    private Map<ClaimMapping, String> getUserAttributesCachedAgainstAuthorizationCode(Object authorizationCode) {
+        Map<ClaimMapping, String> userAttributes = Collections.emptyMap();
+        if (authorizationCode != null) {
+            // get the cached user claims against the authorization code if any
+            userAttributes = getUserAttributesFromCacheUsingCode(authorizationCode.toString());
+        }
+        return userAttributes;
+    }
+
+    private Map<ClaimMapping, String> getUserAttributesCachedAgainstToken(String accessToken) {
+        Map<ClaimMapping, String> userAttributes = Collections.emptyMap();
+        if (accessToken != null) {
+            // get the user claims cached against the access token if any
+            userAttributes = getUserAttributesFromCacheUsingToken(accessToken);
+        }
+        return userAttributes;
+    }
+
+    private Map<String, Object> getUserClaims(OAuthAuthzReqMessageContext authzReqMessageContext)
+            throws OAuthSystemException {
+
+        Map<String, Object> claims = Collections.emptyMap();
+        Map<ClaimMapping, String> userAttributes =
+                getUserAttributesCachedAgainstToken(getAuthorizationCode(authzReqMessageContext));
```

Comment:
**getAuthorizationCode**: method name is invalid. Should be like **getAccessToken**